### PR TITLE
*: take care of anti-MEV part

### PR DESCRIPTION
**Pull request by AnnaShaleva**
1. For anti-MEV extension it's possible that some commits carry invalid share which is impossible to check at the commit verification level. In this case node should collect some more commits. Hence, ProcessBlock callback signature is extended with error return value. The general idea is similar to PreBlock processing behaviour.
2. Remove PreCommit with invalid signature from cache.
3. Log Commit signature verification error.
4. Add VerifyCommit callback.

### CODE REVIEW

**Comment by AnnaShaleva** [comment]
@roman-khimov, review please, but simulation/CHANGELOG check are failing; will check what's happened with simulation.

File: check.go [hunk 1]
```diff
@@ -124,8 +122,17 @@ func (d *DBFT[H]) checkCommit() {
 		zap.Stringer("merkle", d.block.MerkleRoot()),
 		zap.Stringer("prev", d.block.PrevHash()))
 
+	err := d.ProcessBlock(d.block)
+	if err != nil && d.isAntiMEVExtensionEnabled() { // errors not supported for AntiMEVExtensions mode off.
```
Comment (roman-khimov) [inline check.go:126]: That's not obvious for library users, we need to handle them somehow anyway.

Comment (AnnaShaleva) [reply check.go]: Fixed.

**Comment by AnnaShaleva** [comment]
Simulation is fixed by #135; CHANGELOG is updated.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**